pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -813,14 +813,6 @@ func (p *Provisioner) getSpec(cluster *v3.Cluster) (*v3.ClusterSpec, error) {
 		return nil, err
 	}
 
-	// Version is the only parameter that can be updated for EKS, if they is equal we do not need to update
-	// TODO: Replace with logic that is more adaptable
-	if cluster.Spec.GenericEngineConfig != nil && (*cluster.Spec.GenericEngineConfig)["driverName"] == "amazonelasticcontainerservice" &&
-		cluster.Status.AppliedSpec.GenericEngineConfig != nil && (*cluster.Spec.GenericEngineConfig)["kubernetesVersion"] ==
-		(*cluster.Status.AppliedSpec.GenericEngineConfig)["kubernetesVersion"] {
-		return nil, nil
-	}
-
```

Comment:
This must be removed to allow client id and secret to be updated right ? 

For the upgrade-when-downstream-is-already-upgraded scenario, it previously never would have gotten hit because it returned an error above ? Only when the cluster legitimately didn't need to be upgraded. And now in that same scenario the cluster will be refreshed from update call above, so the standard checks will work. That sound right ?

Reply:
It actually still got hit because as far as rancher was concerned the version were not equal. Scenario:
your desired and current version is 1.13.
in eks you change to 1.14.
in rancher you upgrade to 1.14, now current is 1.13 and desired is 1.14.
controller explodes.

This was the same prior and is the second issue this PR addresses.

Reply:
So to confirm new approach in that same scenario: everything is the same except when you upgrade it doesn't error out and status/spec would both equal 1.14. And also client id/secret can be updated.

Reply:
Correct.